zappa/core.py

```diff
@@ -315,13 +316,18 @@ def __init__(
         else:
             self.manylinux_suffix_start = "cp39"
 
+        self.architecture = architecture
+        if not self.architecture:
+            self.architecture = "x86_64"
+
         # AWS Lambda supports manylinux1/2010, manylinux2014, and manylinux_2_24
         manylinux_suffixes = ("_2_24", "2014", "2010", "1")
         self.manylinux_wheel_file_match = re.compile(
-            rf'^.*{self.manylinux_suffix_start}-(manylinux_\d+_\d+_x86_64[.])?manylinux({"|".join(manylinux_suffixes)})_x86_64[.]whl$'  # noqa: E501
+            rf"^.*{self.manylinux_suffix_start}-(manylinux_\d+_\d+_{self.architecture}"
+            rf'[.])?manylinux({"|".join(manylinux_suffixes)})_{self.architecture}[.]whl$'
         )
         self.manylinux_wheel_abi3_file_match = re.compile(
-            rf'^.*cp3.-abi3-manylinux({"|".join(manylinux_suffixes)})_x86_64.whl$'
+            f'^.*cp3.-abi3-manylinux({"|".join(manylinux_suffixes)})_{self.architecture}.whl$'
```

Comment:
Packages use "arm64" and "aarch64" extensions to publish arm architecture-related manylinux wheels.